esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2278,6 +2278,55 @@ bool CFileSprayEx::onReplicate(IEspContext &context, IEspReplicate &req, IEspRep
     return true;
 }
 
+const char* CFileSprayEx::getDropZoneDirByIP(const char* ip, StringBuffer& dir)
+{
+    if (!ip || !*ip)
+        return NULL;
+
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    Owned<IConstEnvironment> env = factory->openEnvironment();
+    if (!env)
+        return NULL;
+
+    Owned<IPropertyTree> root = &env->getPTree();
+    VStringBuffer xPath("Hardware/Computer[@netAddress='%s']/@name", ip);
+    const char* computer = root->queryProp(xPath.str());
+    if (!computer || !*computer)
```

Comment:
questioning whether this should ever happen really.
I realize it can at the moment, by virtue of the user being able to override the IP despite the drop list selection.
(See other comment).